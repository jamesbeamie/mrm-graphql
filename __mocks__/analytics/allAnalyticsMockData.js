export default {
  analytics: {
    bookingsCount: [
      {
        totalBookings: 0,
        period: 'Jan 2018',
      },
      {
        totalBookings: 0,
        period: 'Feb 2018',
      },
      {
        totalBookings: 0,
        period: 'Mar 2018',
      },
      {
        totalBookings: 0,
        period: 'Apr 2018',
      },
      {
        totalBookings: 0,
        period: 'May 2018',
      },
      {
        totalBookings: 0,
        period: 'Jun 2018',
      },
      {
        totalBookings: 0,
        period: 'Jul 2018',
      },
      {
        totalBookings: 0,
        period: 'Aug 2018',
      },
      {
        totalBookings: 0,
        period: 'Sep 2018',
      },
      {
        totalBookings: 0,
        period: 'Oct 2018',
      },
      {
        totalBookings: 0,
        period: 'Nov 2018',
      },
      {
        totalBookings: 0,
        period: 'Dec 2018',
      },
      {
        totalBookings: 0,
        period: 'Jan 2019',
      },
      {
        totalBookings: 0,
        period: 'Feb 2019',
      },
      {
        totalBookings: 0,
        period: 'Mar 2019',
      },
      {
        totalBookings: 0,
        period: 'Apr 2019',
      },
      {
        totalBookings: 0,
        period: 'May 2019',
      },
    ],
    checkinsPercentage: 0.0,
    appBookingsPercentage: 0.0,
    cancellationsPercentage: 0.0,
    autoCancellationsPercentage: 0.0,
    analytics: [
      {
        roomName: 'Kampala',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Kampala2',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'New York',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Cafeteria',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'China Town',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Hogwarts',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Kigali',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Krypton',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Pich',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Rabat',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Rich Port',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Stone Henge',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Room1',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Tent',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Tokyo',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Younde',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
      {
        roomName: 'Valhalla',
        cancellations: 0,
        cancellationsPercentage: 0.0,
        autoCancellations: 0,
        numberOfBookings: 0,
        checkins: 0,
        checkinsPercentage: 0.0,
        bookingsPercentageShare: 0.0,
        appBookings: 0,
        appBookingsPercentage: 0.0,
        events: [
          {
            durationInMinutes: 0,
          },
        ],
      },
    ],
    rooms: [
      {
        id: '6',
        name: 'Cafeteria',
        capacity: 64,
      },
      {
        id: '3',
        name: 'China Town',
        capacity: 4,
      },
      {
        id: '5',
        name: 'Hogwarts',
        capacity: 64,
      },
      {
        id: '18',
        name: 'Kampala',
        capacity: 14,
      },
      {
        id: '4',
        name: 'Kampala2',
        capacity: 4,
      },
      {
        id: '14',
        name: 'Kigali',
        capacity: 30,
      },
      {
        id: '13',
        name: 'Krypton',
        capacity: 30,
      },
      {
        id: '2',
        name: 'New York',
        capacity: 4,
      },
      {
        id: '11',
        name: 'Pich',
        capacity: 50,
      },
      {
        id: '8',
        name: 'Rabat',
        capacity: 14,
      },
      {
        id: '12',
        name: 'Rich Port',
        capacity: 10,
      },
      {
        id: '17',
        name: 'Room1',
        capacity: 6,
      },
      {
        id: '15',
        name: 'Stone Henge',
        capacity: 20,
      },
      {
        id: '7',
        name: 'Tent',
        capacity: 164,
      },
      {
        id: '16',
        name: 'Tokyo',
        capacity: 20,
      },
      {
        id: '9',
        name: 'Valhalla',
        capacity: 14,
      },
      {
        id: '10',
        name: 'Younde',
        capacity: 14,
      },
    ],
  },
  fetching: false,
};
