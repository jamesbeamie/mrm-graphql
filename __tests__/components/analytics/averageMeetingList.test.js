import React from 'react';
import { mount } from 'enzyme';
import AverageMeetingListComponent from '../../../src/components/analytics/AverageMeetingList';
import AnalyticsContext from '../../../src/components/helpers/AnalyticsContext';
import allAnalyticsMockData from '../../../__mocks__/analytics/allAnalyticsMockData';


describe('Average Meeting List Component', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(
      <AnalyticsContext.Provider value={allAnalyticsMockData} >
        <AverageMeetingListComponent />
      </AnalyticsContext.Provider>,
    );
  });

  it('renders correctly from memory', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
