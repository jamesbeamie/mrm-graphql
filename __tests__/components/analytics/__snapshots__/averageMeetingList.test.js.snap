// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Average Meeting List Component renders correctly from memory 1`] = `
<AverageMeetingList>
  <div
    className="average-meeting overlay-container"
  >
    <div
      className="average-meeting-control"
    >
      <h4
        className="header-title"
      >
        Average time spent/Meeting Room
      </h4>
      <span
        className="moreVerticalIcon"
      >
        <ThemedTooltippedComponent
          className=""
          composeTheme="deeply"
          mapThemrProps={[Function]}
          theme={null}
          tooltip="The number of meetings in a room,  the average number of attendees to these meetings as well as the average duration of the meetings."
          tooltipDelay={0}
          tooltipHideOnClick={true}
          tooltipPosition="top"
          tooltipShowOnClick={true}
        >
          <TooltippedComponent
            className=""
            theme={
              Object {
                "tooltip": "tooltip",
                "tooltipInner": "tooltipInner",
              }
            }
            tooltip="The number of meetings in a room,  the average number of attendees to these meetings as well as the average duration of the meetings."
            tooltipDelay={0}
            tooltipHideOnClick={true}
            tooltipPosition="top"
            tooltipShowOnClick={true}
          >
            <ThemedLink
              active={false}
              className=""
              composeTheme="deeply"
              mapThemrProps={[Function]}
              onClick={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              theme={
                Object {
                  "tooltip": "tooltip",
                  "tooltipInner": "tooltipInner",
                }
              }
            >
              <Link
                active={false}
                className=""
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                theme={
                  Object {
                    "tooltip": "tooltip",
                    "tooltipInner": "tooltipInner",
                  }
                }
              >
                <a
                  className=""
                  data-react-toolbox="link"
                  onClick={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <img
                    alt="help icon"
                    src="https://firebasestorage.googleapis.com/v0/b/learning-map-app.appspot.com/o/MrmFrontImages%2Fhelp_outline_24px.svg?alt=media&token=839e3137-c29e-463d-9a68-361e28794911"
                  />
                </a>
              </Link>
            </ThemedLink>
          </TooltippedComponent>
        </ThemedTooltippedComponent>
      </span>
    </div>
    <div
      className="average-meeting-list"
    >
      <div>
        <TableHead
          titles={
            Array [
              "Room",
              "No. of meetings",
              "Average Meeting Duration",
            ]
          }
        >
          <div
            className="table__headers"
          >
            <span
              key="Room"
            >
              Room
            </span>
            <span
              key="No. of meetings"
            >
              No. of meetings
            </span>
            <span
              key="Average Meeting Duration"
            >
              Average Meeting Duration
            </span>
          </div>
        </TableHead>
        <div>
          <QueryAnalyticsPerMeetingRoom
            data={
              Object {
                "hasNext": true,
                "hasPrevious": false,
                "meetingDurationAnalytics": Array [
                  Object {
                    "count": 0,
                    "roomName": "Kampala",
                    "totalDuration": 0,
                  },
                  Object {
                    "count": 0,
                    "roomName": "Kampala2",
                    "totalDuration": 0,
                  },
                  Object {
                    "count": 0,
                    "roomName": "New York",
                    "totalDuration": 0,
                  },
                  Object {
                    "count": 0,
                    "roomName": "Cafeteria",
                    "totalDuration": 0,
                  },
                  Object {
                    "count": 0,
                    "roomName": "China Town",
                    "totalDuration": 0,
                  },
                ],
                "pages": 4,
              }
            }
          >
            <div
              className="table__row--analytics"
              key="1"
            >
              <span>
                Kampala
              </span>
              <span>
                0
              </span>
              <span>
                0 Hour 0 Minute
              </span>
            </div>
            <div
              className="table__row--analytics"
              key="2"
            >
              <span>
                Kampala2
              </span>
              <span>
                0
              </span>
              <span>
                0 Hour 0 Minute
              </span>
            </div>
            <div
              className="table__row--analytics"
              key="3"
            >
              <span>
                New York
              </span>
              <span>
                0
              </span>
              <span>
                0 Hour 0 Minute
              </span>
            </div>
            <div
              className="table__row--analytics"
              key="4"
            >
              <span>
                Cafeteria
              </span>
              <span>
                0
              </span>
              <span>
                0 Hour 0 Minute
              </span>
            </div>
            <div
              className="table__row--analytics"
              key="5"
            >
              <span>
                China Town
              </span>
              <span>
                0
              </span>
              <span>
                0 Hour 0 Minute
              </span>
            </div>
          </QueryAnalyticsPerMeetingRoom>
        </div>
      </div>
    </div>
    <div
      className="average-meeting-pagination"
    >
      <div>
        <Pagination
          currentPage={1}
          dataFetched={true}
          handleData={[Function]}
          hasNext={true}
          hasPrevious={false}
          isFetching={false}
          itemsPerPage={
            Array [
              5,
              10,
              20,
              50,
            ]
          }
          perPage={5}
          reverse={false}
          totalPages={4}
        >
          <nav
            className="page-navigation"
          >
            <div
              className="perPageBlock"
            >
              <ul
                className="pagination"
              >
                <select
                  className="page-select"
                  disabled={false}
                  id="perPage"
                  name="perPage"
                  onChange={[Function]}
                  value={5}
                >
                  <option
                    key="6"
                  >
                    5
                  </option>
                  <option
                    key="11"
                  >
                    10
                  </option>
                  <option
                    key="21"
                  >
                    20
                  </option>
                  <option
                    key="51"
                  >
                    50
                  </option>
                </select>
                <li>
                  Items per page
                </li>
              </ul>
            </div>
            <div
              className="showingBlock"
            >
              <ul
                className="pagination"
              >
                <li>
                  <button
                    className="disabled"
                    id="previous"
                    onClick={[Function]}
                  >
                    Previous
                  </button>
                </li>
                <li
                  className="results"
                >
                  <ul>
                    <span>
                      Showing 
                      page
                    </span>
                    <select
                      className="totalPage"
                      disabled={false}
                      id="totalPage"
                      name="page"
                      onChange={[Function]}
                      value={1}
                    >
                      <option
                        key="0"
                      >
                        1
                      </option>
                      <option
                        key="1"
                      >
                        2
                      </option>
                      <option
                        key="2"
                      >
                        3
                      </option>
                      <option
                        key="3"
                      >
                        4
                      </option>
                    </select>
                    <span
                      className="pageNum"
                    >
                       of 
                      4
                    </span>
                  </ul>
                </li>
                <li>
                  <button
                    className="enabled"
                    id="next"
                    onClick={[Function]}
                  >
                    Next
                  </button>
                </li>
              </ul>
            </div>
          </nav>
        </Pagination>
      </div>
    </div>
  </div>
</AverageMeetingList>
`;
